### ISSUE TITLE
Typst support

### ISSUE THREAD

**Issue by Atreyu-94** [textext/textext#403]
Would it be possible for you to support Typst? It has priority support for svg and pdf.

https://typst.app/docs/

**Comment by jcwinkler** [comment]
Yes, this should be easy to implement. I will try this. Nice idea!

(Later we can decide if this should go into a separate extension because the name TexText does not lead intuitively to Typst)

**Comment by Atreyu-94** [comment]
I think it's a great idea to group them all within the same extension. Maybe making a clarification in the readme.
I've been using this extension for a long time and compatibility with Typst would be great :D

**Comment by jcwinkler** [comment]
ToDo for TexText 1.10:

- [ ] If typst is selected: disable preamble file
- [ ] TeX command --> Compiler command (pdflatex, typst, ...)
- [ ] Update documenation

ToDo TexText 2.0
- [ ] typst is able to produce svg code directly. Hence we do not need pdf -> svg conversion for typst

**Comment by cknoll** [comment]
I can confirm that typst support works on my system (debian 11). Because I do not have GTK support installed it automatically opens in TkInter-mode.


![Screenshot_20231026_121147](https://github.com/textext/textext/assets/1315698/c5915b0c-b35f-4cf0-98d9-17784f4d688d)

**Comment by jcwinkler** [comment]
This is implemented now in release 1.10.0.

**Comment by OpenFoam-User** [comment]
@jcwinkler Hi, thank you for adding typst support.

I have downloaded the exe version of textext and  installed it on Windows 11. I am using  `Inkscape 1.4-dev (4ca261c, 2023-11-03, custom)`.

When I open the textext editor, I do not see typst listed in list of compilers despite being added to System Path.

![image](https://github.com/textext/textext/assets/67183566/15b9ec1b-5217-4e0a-8802-05200e24b7e3)


**P.S.** When I run the following command on cmd or powershell: 

```
typst --version
typst 0.8.0 (360cc9b9)
```

**Comment by YDX-2147483647** [comment]
> When I run the following command on cmd or powershell

@OpenFoam-User Are you on Windows? `setup_win.bat` may have explained the reason.

```log
[TexText][SUCCESS ]: /-and-+ [Succ] Detect *latex
[TexText][SUCCESS ]: |     /--or-* [Succ] `pdflatex.exe` is found at `…`
…
[TexText][INFO    ]: |     /--or-* [Fail] `typst` is NOT found in PATH
[TexText][INFO    ]: |            Please follow installation instructions at
[TexText][INFO    ]: |                   https://textext.github.io/textext/install/windows.html#windows-install-preparation
[TexText][INFO    ]: |            If typst is installed in custom location, specify it via
[TexText][INFO    ]: |                   --typst-executable=<path-to-typst>
[TexText][INFO    ]: |            and run setup.py again
```

If so, rerun the script with `--typst-executable`, and it'll work.

```powershell
> .\setup_win.bat --typst-executable C:/path/to/typst.exe
# Alternatively, …
> .\setup_win.bat --typst-executable (Get-Command typst).Source
```

(I guess the reason is `typst.exe` ≠ `typst`)

**Comment by OpenFoam-User** [comment]
@YDX-2147483647 : Many thanks! that solved it. I have skimmed quickly through the batch script but missed that flag to specify the path to typst explicity.

On thing though is that the `Preview` button is grayed out. Only the `Save` button works.

![image](https://github.com/textext/textext/assets/67183566/33033043-5cdd-4c2b-b530-1c54c57bfdd4)

**Comment by jcwinkler** [comment]
This is intended. Preview is not implemented for typst - as stated in the documentation. It will be available in TexText 2.0.

**Comment by jcwinkler** [comment]
TexText 2.0 will provide a preamble file for typst.
